public/app/core/services/context_srv.ts

```diff
@@ -133,18 +126,32 @@ export class ContextSrv {
     }
   }
 
+  hasAccessToExplore() {
+    return this.hasPermission(AccessControlAction.DataSourcesExplore) && config.exploreEnabled;
+  }
+
+  // evaluates access control permissions, granting access if the user has any of them
+  evaluatePermission(actions: string[]) {
+    if (this.authorizer.hasAnyPermission(this.user, actions)) {
+      return [];
+    }
+    // Hack to reject when user does not have permission
+    // TODO(aarongodin): fix this hack?
+    return ['Reject'];
```

Comment:
yeah, here's we return string value which then compared to the basic role of the use. So 'Reject' won't match any basic role, therefore access will be denied.

Reply:
That's fair, just seemed very odd so I wanted to get a TODO in there. I'll spend some time on it soon and see if I can make it a little more obvious.